opam install jbuilder fails if $MAKEFLAGS is set

```
echo $MAKEFLAGS
-j7
```

```
=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of jbuilder failed at "./boot.exe -j 7".

#=== ERROR while installing jbuilder.1.0+beta19 ===============================#
# opam-version 1.2.2
# os           linux
# command      ./boot.exe -j 7
# path         /home/jan/.opam/4.05.0/build/jbuilder.1.0+beta19
# compiler     4.05.0
# exit-code    1
# env-file     /home/jan/.opam/4.05.0/build/jbuilder.1.0+beta19/jbuilder-16203-d0f245.env
# stdout-file  /home/jan/.opam/4.05.0/build/jbuilder.1.0+beta19/jbuilder-16203-d0f245.out
# stderr-file  /home/jan/.opam/4.05.0/build/jbuilder.1.0+beta19/jbuilder-16203-d0f245.err
### stderr ###
# Description: ("Env.of_unix: duplicated variable found in the environment"
# [...]
#  (value1 "")
#  (value2 -j7))
# Backtrace:
# Raised at file "src/errors.ml", line 17, characters 31-45
# Called from file "src/env.ml", line 61, characters 12-54
# Called from file "camlinternalLazy.ml", line 27, characters 17-27
# Re-raised at file "camlinternalLazy.ml", line 34, characters 10-11
# Called from file "src/context.ml", line 341, characters 43-59
# Called from file "src/fiber/fiber.ml", line 312, characters 4-14
```

--- a/src/env.ml
+++ b/src/env.ml
@@ -44,15 +44,10 @@ let of_unix arr =
       Sexp.code_error "Env.of_unix: entry without '=' found in the environ"
         ["var", Sexp.To_sexp.string s]
     | Some (k, v) -> (k, v))
-  |> Map.of_list
-  |> function
-  | Ok x -> x
-  | Error (var, v1, v2) ->
-    Sexp.code_error "Env.of_unix: duplicated variable found in the environment"
-      [ "var"   , Sexp.To_sexp.string var
-      ; "value1", Sexp.To_sexp.string v1
-      ; "value2", Sexp.To_sexp.string v2
-      ]
+  |> Map.of_list_multi
+  |> Map.map ~f:(function
+    | [] -> assert false
+    | x::_ -> x)
 
 let initial =
   let i =

